### ISSUE TITLE
ByteCode definition leading to large "bytes" ?

### ISSUE THREAD

**Issue by schtauffen** [jonnyboyC/spacelox#1]
Hello! First I would like to thank you for having this repository. It has been invaluable as I work through crafting interpreters with rust.  

In your definition for bytecode you allow some enums to have values, but this will cause all the enums to have that size: 
 https://github.com/jonnyboyC/spacelox/blob/0808b0138e399539d28e30feef9baef0cc5c34e4/src/chunk.rs#L46

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0a47a8a24b4c8c33e01c93001bd05ecf

Was this intentional?

**Comment by jonnyboyC** [comment]
Well I'm not too far ahead of you at this point, but glad it's been a good reference! 

Essentially I choose to just place it in the enum as a simplification. I know that rust's enums are all as large as the largest variant so I'm paying the extra byte for each instruction (not ideal). I playing around with other ideas but didn't want to get too bogged down coming up with a more efficient solution while also learning the material in the book. 

So long story short it was intentionally, but if I was better versed in rust I'm sure there is a better solution.

**Comment by schtauffen** [comment]
It is much more simple that way, so I appreciate using it for simplicity.
My other ask, if you don't mind me asking, is if the current implementation of "free_objects" does anything / is necessary? I don't know enough rust to know if this is freeing the memory somehow by passing ownership to the noop which then will get it deallocated?

I have added my implementation if you are curious:
https://github.com/schtauffen/rslox

I am currently trying to figure out string internment but it seems more difficult than in C.

**Comment by jonnyboyC** [comment]
I _believe_ that it's really not necessary. I mostly left it in place if I encountered something later in the book that may actually need it. My understanding is that once a `Value` is removed from the the vm's stack it and its children will be free'd potentially running the `Drop` trait if it's implemented. 

I'll definitely check out your implementation. I've been doing the same with other rust impls to get ideas how to create various features.
